dist/DeadlineCloudSubmitter.jsx

```
@@ -2263,38 +2299,96 @@ function buildUI(thisObj) {
+    settingsGroup.alignChildren = ['left', 'top'];
 
-    var framesPerTaskGroup = controlsPanel.add("group", undefined, "");
+    // Setting up frame per task GUI
+    const framesPerTaskGroup = settingsGroup.add("group", undefined, "");
     framesPerTaskGroup.orientation = "row";
     framesPerTaskGroup.alignment = ['fill', 'top'];
-    framesPerTaskGroup.alignChildren = ['left', 'top'];
+    framesPerTaskGroup.alignChildren = ['left', 'center'];
 
-    var framesPerTaskLabel = framesPerTaskGroup.add("statictext", undefined, "Frames per task");
+    const framesPerTaskLabel = framesPerTaskGroup.add("statictext", undefined, "Frames per task");
     framesPerTaskLabel.alignment = ['left', 'center'];
     framesPerTaskLabel.helpTip = "The number of frames per task. Only affects image sequence output."
 
     const framesPerTaskTextBox = framesPerTaskGroup.add("edittext", undefined, "");
     framesPerTaskTextBox.alignment = ['fill', 'top'];
     framesPerTaskTextBox.helpTip = framesPerTaskLabel.helpTip;
     framesPerTaskTextBox.onChange = function() {
-        const newFramesPerTaskValue = String(Math.abs(parseInt(framesPerTaskTextBox.text)));
-        if (newFramesPerTaskValue == "NaN") {
-            framesPerTaskTextBox.text = persistentFramesPerTask;
-        }
-        if (Math.abs(parseInt(newFramesPerTaskValue) > 9999)) {
+        const newFramesPerTaskValue = Math.abs(parseInt(framesPerTaskTextBox.text));
+        if (isNaN(newFramesPerTaskValue)) {
+            framesPerTaskTextBox.text = app.settings.getSetting(DEADLINECLOUD_SUBMITTER_SETTINGS, DEADLINECLOUD_FRAMESPERTASK);
+        } else if (newFramesPerTaskValue > 9999) {
             framesPerTaskTextBox.text = "9999";
+        } else {
+            // Need to reassign in case input string is a number followed my random characters
+            // since parseInt parses the first number it finds in a provided string.
+            framesPerTaskTextBox.text = newFramesPerTaskValue;
         }
```

If possible, I think we should have an error message that shows up if the entered value is incorrect, rather than silently falling back to the last known value.

Also, I don't think we should treat a value with text in it (e.g. 12t) as input for 12. This is probably a typo on the part of the user, with a good chance that 12 wasn't actually the intended value, so we should reject the input entirely.

---

That's a good point, but should probably be addressed in a separate PR, since this portion of the code is just a refactor to make the code clearer and also make it clear the the customer what "10d" was being evaluated to by our existing code. Originally, "10d" would've been interpreted as "10" by our mainline code, but the customer would have had "10d" saved to their settings, which is weird user experience. Thoughts?

---

Sure, if this is the current experience I am fine with improving this in a followup.